src/Services/RetrieveExternalDataService/RetrieveExternalDataService.cs

```diff
@@ -65,15 +66,25 @@ public async Task Process(List<IAction> actions, FormSchema formSchema, string f
                 var response = entity.IsPost ? await _gateway.PostAsync(entity.Url, mappingData.Data) :
                         await _gateway.GetAsync(entity.Url);
 
-                if (!response.IsSuccessStatusCode)
-                    throw new ApplicationException($"RetrieveExternalDataService::Process, http request to {entity.Url} returned an unsuccessful status code, Response: {JsonConvert.SerializeObject(response)}");
-
                 var responseAnswer = string.Empty;
-                if (response.Content is not null)
+                if (response.StatusCode.Equals(HttpStatusCode.NotFound))
                 {
+                    responseAnswer = null;
+                }
+                else if (response.IsSuccessStatusCode)
+                {
+                    if (response.Content is null)
+                        throw new ApplicationException($"RetrieveExternalDataService::Process, http request to {entity.Url} returned an unsuccessful status code, Response: {JsonConvert.SerializeObject(response)}");
+
                     string content = await response.Content.ReadAsStringAsync();
-                    if (!string.IsNullOrEmpty(content))
-                        responseAnswer = System.Text.Json.JsonSerializer.Deserialize<string>(content);
+                    if (string.IsNullOrEmpty(content))
+                        throw new ApplicationException($"RetrieveExternalDataService::Process, http request to {entity.Url} returned an unsuccessful status code, Response: {JsonConvert.SerializeObject(response)}");
```

Comment:
Error message says returned an unsuccessful status code, should this be content was null or empty?

Reply:
Indeed